### ISSUE TITLE
Problems with saving models

### ISSUE THREAD

**Issue by jzyxn** [Zyun-Y/DconnNet#23]
Hi, saving the model the way you did in your code, why is it that when loading the trained model using "test_only", it has the same effect as randomly initialising the model, with all metrics being close to 0 (on the ISIC2018 dataset), except for accuracy?

**Comment by Zyun-Y** [comment]
Did you load the model by setting the "--pretrained" argument?

**Comment by jzyxn** [comment]
Thanks for the reply, I've found out why I couldn't save the model, I used 
with torch.cuda.amp.autocast() instead of
net, optimizer = amp.initialize(model, optim, opt_level='O2') 
for training, but I can't reasonably save the model as well as load the model (all metrics are close to 0 except accuracy) with torch.cuda.amp.autocast(), and if without this statement, then I can reasonably save, load the model
net, optimizer = amp.initialize(model, optim, opt_level='O2') Do you get the same error using this net, optimizer = amp.initialize(model, optim, opt_level='O2')? And do you know how to save, load model using torch.cuda.amp.autocast()?

**Comment by Zyun-Y** [comment]
Glad you figured it out. Unfortunately, I didn't use torch.cuda.amp.autocast() in this code so maybe you can find a solution in their [official documentation](https://pytorch.org/tutorials/recipes/recipes/amp_recipe.html).

**Comment by jzyxn** [comment]
OK，thanks
